vms/rpcchainvm/ghttp/http_client.go

```diff
@@ -156,9 +156,17 @@ func (c *Client) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	_, err = c.client.Handle(r.Context(), req)
+	reply, err := c.client.Handle(r.Context(), req)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	header := w.Header()
+	clear(header)
+
+	for _, h := range reply.Header {
+		header[h.Key] = h.Values
 	}
```

Comment:
I think we also have a helper for this right now: `grpcutils.MergeHTTPHeader` That implementation doesn't clear the input header... But I think it should. Currently if the implementation deletes a header that is not honored.

Reply:
Updated, + added coverage on both http + http simple